client/rpc-api/src/chain/mod.rs

```diff
@@ -18,4 +18,58 @@
 
 //! Substrate blockchain API.
 
+use jsonrpsee::{proc_macros::rpc, types::JsonRpcResult};
+use sp_rpc::{list::ListOrValue, number::NumberOrHex};
+
 pub mod error;
+
+#[rpc(client, server, namespace = "chain")]
```

Comment:
Why both client and server?

Reply:
It is used to mimic https://github.com/paritytech/substrate/blob/master/client/rpc-api/src/child_state/mod.rs#L25

So, these can used to generate a client or have the trait automatically implemented, such as https://github.com/paritytech/substrate/blob/master/bin/node/rpc-client/src/main.rs